docs/source/user-guide/vision/quickstarts/plot_detection_tutorial.py

```diff
@@ -224,114 +175,135 @@ def show(imgs):
+    corresponding to a sample. Each array element is in a shape of [B, 5], where B is the number of bboxes
+    in the image, and each bounding box is in the structure of [class_id, x, y, w, h].
+    """
+    tensor_annotations = batch[1]
+    label = []
+    for annotation in tensor_annotations:
+        if len(annotation["boxes"]):
+            bbox = torch.stack(annotation["boxes"])
+            # Convert the Pascal VOC xyxy format to xywh format
+            bbox[:, 2:] = bbox[:, 2:] - bbox[:, :2]
+            # The label shape is [class_id, x, y, w, h]
+            label.append(
+                torch.concat([torch.stack(annotation["labels"]).reshape((-1, 1)), bbox], dim=1)
+            )
+        else:
+            # If it's an empty image, we need to add an empty label
+            label.append(torch.tensor([]))
+    return label
+
+def infer_on_batch(batch):
+    """
+    Returns the predictions for a batch of data. The expected format is an iterator of arrays, each array
+    corresponding to a sample. Each array element is in a shape of [B, 6], where B is the number of bboxes in the
+    predictions, and each bounding box is in the structure of [x, y, w, h, score, class_id].
+
+    Note that model and device here are global variables, and are defined in the previous code block, as the collate
+    function cannot recieve other arguments than the batch.
+    """
+    nm_thrs = 0.2
+    score_thrs = 0.7
+    imgs = list(img.to(device) for img in batch[0])
+    # Getting the predictions of the model on the batch
+    with torch.no_grad():
+        preds = model(imgs)
+    processed_pred = []
+    for pred in preds:
+        # Performoing non-maximum suppression on the detections
+        keep_boxes = torchvision.ops.nms(pred['boxes'], pred['scores'], nm_thrs)
+        score_filter = pred['scores'][keep_boxes] > score_thrs
+
+        # get the filtered result
+        test_boxes = pred['boxes'][keep_boxes][score_filter].reshape((-1, 4))
+        test_boxes[:, 2:] = test_boxes[:, 2:] - test_boxes[:, :2]  # xyxy to xywh
+        test_labels = pred['labels'][keep_boxes][score_filter]
+        test_scores = pred['scores'][keep_boxes][score_filter]
+
+        processed_pred.append(
+            torch.concat([test_boxes, test_scores.reshape((-1, 1)), test_labels.reshape((-1, 1))], dim=1))
+    return processed_pred
 
 #%%
-# After defining the task class, we can validate it by running the following code:
+# Now we'll create the collate function that will be used by the DataLoader.
+# In pytorch, the collate function is used to transform the output batch to any custom format, and we'll use that
+# in order to transform the batch to the correct format for the checks.
+
+def deepchecks_collate_fn(batch):
+    """Return a batch of images, labels and predictions in the expected format."""
+    batch = tuple(zip(*batch))
+    images = batch_to_images(batch)
+    labels = batch_to_labels(batch)
+    predictions = infer_on_batch(batch, model, device)
```

Comment:
I think will be clearer if each util function will run only on the relevant data - batch_to_images(images), ...
would also change the function names to distiguise it from the previous version where it run on a batch